config/config.go

```diff
@@ -29,186 +29,243 @@ import (
 )
 
 const (
-	// ServiceNameKey is the config key of the service name
+	// ServiceNameKey is the config key of the service name.
 	ServiceNameKey = "name"
+
 	// ServiceDescriptionKey is the config key of the service
-	// description
+	// description.
 	ServiceDescriptionKey = "description"
-	// ServiceOwnerKey is the config key for a service owner
+
+	// ServiceOwnerKey is the config key for a service owner.
 	ServiceOwnerKey = "owner"
 )
 
 const (
-	_appRoot     = "APP_ROOT"
+	_appRoot     = "_ROOT"
 	_environment = "_ENVIRONMENT"
 	_configDir   = "_CONFIG_DIR"
 	_configRoot  = "./config"
 	_baseFile    = "base"
 	_secretsFile = "secrets"
+	_devEnv      = "development"
 )
 
-var (
-	_setupMux sync.Mutex
+type lookUpFunc func(string) (string, bool)
 
-	_envPrefix            = "APP"
-	_staticProviderFuncs  = []ProviderFunc{YamlProvider()}
-	_configFiles          = baseFiles()
-	_dynamicProviderFuncs []DynamicProviderFunc
-)
+// Loader is responsible for loading configs.
+type Loader struct {
+	lock sync.RWMutex
 
-var (
-	_devEnv = "development"
-)
+	envPrefix            string
+	staticProviderFuncs  []ProviderFunc
+	dynamicProviderFuncs []DynamicProviderFunc
+
+	// Files to load.
+	configFiles []string
+
+	// Dirs to load from.
+	dirs []string
+
+	// Where to look for environment variables.
+	lookUp lookUpFunc
+}
+
+// DefaultLoader will be used if config is not defined for a service.
+var DefaultLoader = newDefaultLoader()
+
+func newDefaultLoader() *Loader {
```

Comment:
I realized that my both comments are conflicting for/against global objects. I prefer having global variable and exported `New` simply for accessibility, but either way is fine.